### PR TITLE
ENH: make `savgol_coeffs`, `savgol_filter` work for even window length

### LINKED ISSUE #13898: Savitzky-Golay filter for even number data
Savitzky-Golay filter in scipy.signal.savgol_filter implements only for odd numbers of window-lengths but there is substantial amount of literature on even number data.

Reference: 
Jianwen Luo, Kui Ying, Jing Bai, (2005), Savitzky–Golay smoothing and differentiation filter for even number data, Signal Processing, Volume 85, Issue 7, Pages 1429-1434, ISSN 0165-1684, https://doi.org/10.1016/j.sigpro.2005.02.002.

### PR DESCRIPTION
**Pull request by czgdp1807**
#### Reference issue

Closes https://github.com/scipy/scipy/issues/13898

#### Additional information

Please let me know if there are any examples to test `savgol_filter` on even window lengths.
I will updated documentation once I complete testing `savgol_filter`. `savgol_coeffs` work correctly with even window length.

Reference - https://www.researchgate.net/publication/220227041_Savitzky-Golay_smoothing_and_differentiation_filter_for_even_number_data

### CODE REVIEW

**Comment by czgdp1807** [comment]
@ilayn @larsoner Please review this PR. I have added tests from the reference in the description. The change is simple and the coefficients appear to be correct.

**Comment by larsoner** [comment]
@czgdp1807 sorry for the slow response, looks good to me. Could you either merge with upstream/master and push here, rebase against upstream/master, or give maintainers edit rights so I can do it for you? I'd like this to run on the latest version of our CIs before merge. If you can do it in the next couple of days, we should be able to get this into 1.8.0 which branches Monday.

**Comment by czgdp1807** [comment]
> Could you either merge with upstream/master and push here, rebase against upstream/master

Done. Let me know if anything else is required.

**Review by tylerjereddy — APPROVED** [review]
Ok, CI is passing with 100 % patch diff coverage and that was the only remaining core developer review concern here, so I'll merge thanks.

No old tests were modified either, only new tests added.

**Pull request merged**
